### PR TITLE
DDF-4598 Add onClick support to card.tsx

### PR DESCRIPTION
**Pull request by adimka**
#### Who is reviewing it? 

@samuelechu @zta6 @brianfelix 
#### Ask 2 committers to review/merge the PR and tag them here.

@djblue
@vinamartin

#### What are the relevant tickets?
For GH Issues:
Fixes: #4598 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by adimka** [comment]
build now

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/card/card.tsx [hunk 1]
```diff
@@ -55,12 +55,20 @@ interface Props {
   footer: React.ReactNode
   className?: string
   style?: React.CSSProperties
+  onClick?: (e: any) => void
```
Comment (andrewkfiedler) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/card/card.tsx]:
```suggestion
  onClick?: (e: React.MouseEvent) => void
```

**Comment by adimka** [comment]
build now

**Comment by zta6** [comment]
🎉 Hero Successful 🎉 
Local build successful
built downstream project that relies on this change, verified functionality in downstream project.

**Pull request merged**